core/lib/api.ts

```diff
@@ -0,0 +1,9 @@
+import { initTsrReactQuery } from "@ts-rest/react-query/v5";
+
+import { siteApi } from "contracts";
+
+import { env } from "./env/env.mjs";
+
+export const client = initTsrReactQuery(siteApi, {
+	baseUrl: env.NEXT_PUBLIC_PUBPUB_URL,
```

Comment:
I suggest we don't use a public env var, and just rely on the window url

```suggestion
	baseUrl: typeof window === "undefined" ? env.PUBPUB_URL : window.location.origin, 
```

this is slightly more jank, and i'm not sure the `env.PUBPUB_URL` will transfer properly or will even ever get used, but this does work. we could also just make it `? "/" : ...` or something instead of `env.PUBPUB_URL`